src/emulator/auth/operations.ts

```diff
@@ -2138,6 +2265,109 @@ function handleIdpSignUp(
   };
 }
 
+type MfaEnrollment = Schemas["GoogleCloudIdentitytoolkitV1MfaEnrollment"];
+
+interface MfaPendingCredential {
+  _AuthEmulatorMfaPendingCredential: string;
+  localId: string;
+  signInProvider: string;
+}
+
+function mfaPending(
+  state: ProjectState,
+  user: UserInfo,
+  signInProvider: string
+): { mfaPendingCredential: string; mfaInfo: MfaEnrollment[] } {
+  if (!user.mfaInfo) {
+    throw new Error("Internal assertion error: mfaPending called on user without MFA.");
+  }
+  const pendingCredentialPayload: MfaPendingCredential = {
+    _AuthEmulatorMfaPendingCredential: "DO NOT MODIFY",
+    localId: user.localId,
+    signInProvider,
+  };
+
+  // Encode pendingCredentialPayload using base64. We don't encrypt or sign the
+  // data in the Auth Emulator but just trust developers not to modify it.
+  const mfaPendingCredential = Buffer.from(
+    JSON.stringify(pendingCredentialPayload),
+    "utf8"
+  ).toString("base64");
+
+  return { mfaPendingCredential, mfaInfo: user.mfaInfo.map(redactMfaInfo) };
+}
+
+function redactMfaInfo(mfaInfo: MfaEnrollment): MfaEnrollment {
+  return {
+    displayName: mfaInfo.displayName,
+    enrolledAt: mfaInfo.enrolledAt,
+    mfaEnrollmentId: mfaInfo.mfaEnrollmentId,
+    phoneInfo: mfaInfo.unobfuscatedPhoneInfo
+      ? obfuscatePhoneNumber(mfaInfo.unobfuscatedPhoneInfo)
```

Comment:
Clarification question: I see that there's some [branching logic](https://source.corp.google.com/piper///depot/google3/java/com/google/identity/boq/idaas/mfa/MfaUtils.java;l=120-129;rcl=395089263) between unobfuscated vs obfuscated phone numbers - what was the reasoning behind defaulting to obfuscated numbers?

Reply:
From public docs: Normally this will show the phone number associated with this enrollment. In some situations, such as after a first factor sign in, it will only show the obfuscated version of the associated phone number.